### ISSUE TITLE
Concurrent issues caused by a large number of Remote Writes

### ISSUE THREAD

**Issue by zhongzc** [GreptimeTeam/greptimedb#3249]
### What type of bug is this?

Other

### What subsystems are affected?

Storage Engine

### Minimal reproduce step

I used [prometheus-benchmark](https://github.com/VictoriaMetrics/prometheus-benchmark/) to test the performance of the greptimedb remote write, and when I adjusted to high concurrency, DDLs could hardly be executed successfully.

## Prerequisite
[helm](https://helm.sh/), [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl-linux/), k8s cluster

## Run greptimedb
```sh
$ cat config.toml
[http]
addr = "0.0.0.0:4000"

$ cargo run --release -- standalone start -c config.toml
```

## Run bench (targetsCount = 2)
```sh
$ git clone https://github.com/VictoriaMetrics/prometheus-benchmark/
```

With modified `chart/values.yaml`:
```yaml
vmtag: "v1.82.1"
nodeSelector: {}
targetsCount: 2
scrapeInterval: 10s
queryInterval: 10s
scrapeConfigUpdatePercent: 1
scrapeConfigUpdateInterval: 10m
writeConcurrency: 0
writeReplicas: 1
writeReplicaMem: "4Gi"
writeReplicaCPU: 2
remoteStorages:
  gt:
    writeURL: "http://172.17.0.1:4000/v1/prometheus/write"
```

Run
```sh
$ make install
```
 
Errors reported by vmagent, said `unknown columns`:
```sh
$ kubectl -n vm-benchmark logs my-bench-prometheus-benchmark-vmagent-gt-replica-0-6rndsc vmagent
500; response body="{\"error\":\"Invalid request to region 4398046511104(1024, 0), reason: unknown columns: [\\\"address\\\", \\\"broadcast\\\", \\\"operstate\\\", \\\"duplex\\\"]\"}"; re-sending the block in 16.000 seconds
500; response body="{\"error\":\"Invalid request to region 4398046511104(1024, 0), reason: unknown columns: [\\\"mountpoint\\\", \\\"fstype\\\"]\"}"; re-sending the block in 8.000 seconds
500; response body="{\"error\":\"Invalid request to region 4398046511104(1024, 0), reason: unknown columns: [\\\"domainname\\\", \\\"nodename\\\", \\\"release\\\", \\\"machine\\\", \\\"sysname\\\", \\\"version\\\"]\"}"; re-sending the block in 2.000 seconds
500; response body="{\"error\":\"Invalid request to region 4398046511104(1024, 0), reason: unknown columns: [\\\"mountpoint\\\", \\\"fstype\\\"]\"}"; re-sending the block in 32.000 seconds
500; response body="{\"error\":\"Invalid request to region 4398046511104(1024, 0), reason: unknown columns: [\\\"fstype\\\", \\\"mountpoint\\\"]\"}"; re-sending the block in 4.000 seconds
500; response body="{\"error\":\"Invalid request to region 4398046511104(1024, 0), reason: unknown columns: [\\\"cause\\\"]\"}"; re-sending the block in 16.000 seconds
500; response body="{\"error\":\"Invalid request to region 4398046511104(1024, 0), reason: unknown columns: [\\\"version\\\", \\\"release\\\", \\\"sysname\\\", \\\"nodename\\\", \\\"domainname\\\", \\\"machine\\\"]\"}"; re-sending the block in 60.000 seconds
500; response body="{\"error\":\"Invalid request to region 4398046511104(1024, 0), reason: unknown columns: [\\\"mountpoint\\\", \\\"fstype\\\"]\"}"; re-sending the block in 60.000 seconds
500; response body="{\"error\":\"Invalid request to region 4398046511104(1024, 0), reason: unknown columns: [\\\"fstype\\\", \\\"mountpoint\\\"]\"}"; re-sending the block in 60.000 seconds
500; response body="{\"error\":\"Invalid request to region 4398046511104(1024, 0), reason: unknown columns: [\\\"mountpoint\\\", \\\"fstype\\\"]\"}"; re-sending the block in 8.000 seconds
```

Finish
```sh
$ make delete
```

## Run bench (targetsCount = 1000)

With modified `chart/values.yaml`:
```yaml
vmtag: "v1.82.1"
nodeSelector: {}
targetsCount: 1000
scrapeInterval: 10s
queryInterval: 10s
scrapeConfigUpdatePercent: 1
scrapeConfigUpdateInterval: 10m
writeConcurrency: 0
writeReplicas: 1
writeReplicaMem: "4Gi"
writeReplicaCPU: 2
remoteStorages:
  gt:
    writeURL: "http://172.17.0.1:4000/v1/prometheus/write"
```

Run
```sh
$ make install
```

greptimedb kept logging `table does not exist`:
```
...
Logical table `greptime.public.node_network_name_assign_type` does not exist, try creating table
Logical table `greptime.public.node_sockstat_UDP6_inuse` does not exist, try creating table
Logical table `greptime.public.node_boot_time_seconds` does not exist, try creating table
Logical table `greptime.public.node_memory_MemTotal_bytes` does not exist, try creating table
Logical table `greptime.public.node_sockstat_TCP_inuse` does not exist, try creating table
Logical table `greptime.public.node_network_transmit_queue_length` does not exist, try creating table
Logical table `greptime.public.node_memory_DirectMap2M_bytes` does not exist, try creating table
...
```

Finish
```sh
$ make delete
```

### What did you expect to see?

During periods of high concurrency, data can be written normally with remote write, and tables can be created without issues.

### What did you see instead?

greptimedb didn't work normally.

### What operating system did you use?

Ubuntu 22.04 x64

### What version of GreptimeDB did you use?

3201aea3602b68bbd6cbdbb6d05f620517d687bc

**Comment by zhongzc** [comment]
cc @fengjiachun @MichaelScofield

**Comment by fengjiachun** [comment]
See #3257

**Comment by zhongzc** [comment]
fixed by #3257
